_posts/en/newsletters/2019-03-19-newsletter.md

```diff
@@ -0,0 +1,277 @@
+---
+title: 'Bitcoin Optech Newsletter #38'
+permalink: /en/newsletters/2019/03/19/
+name: 2019-03-19-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter gives an update on the planned removal of BIP61
+reject messages from Bitcoin Core, links to further discussion about
+SIGHASH_NOINPUT_UNSAFE, announces some new features in the Esplora block
+chain explorer and summarizes a discussion about its new privacy
+analysis features, provides information about an updated node ban list,
+and links to videos of talks at the recent MIT Bitcoin Club Expo.  Also
+provided are a new weekly section about adoption of bech32 sending
+support and the normal list of notable changes in popular Bitcoin
+infrastructure projects.
+
+## Action items
+
+- **Help test Bitcoin Core 0.18.0 RC2:** The second Release Candidate
+  (RC) for the next major version of Bitcoin Core has been [released][0.18.0].
+  Testing is still needed by organizations and experienced users who
+  plan to run the new version of Bitcoin Core in production.
+
+## News
+
+- **BIP61 reject messages:** as summarized in [last week's
+  newsletter][Newsletter #37], several developers complained about
+  [BIP61][] reject messages being disabled by default in upcoming
+  Bitcoin Core 0.18.0, especially since notice of this was only given
+  shortly before that planned release.  Bitcoin Core developers
+  [discussed][core dev irc] the issue and decided to re-enabled BIP61 by
+  default for the 0.18.0 release but describe it as deprecated in the
+  release notes.  They plan to disable it by default in 0.19.0 (expected
+  around October 2019) and potentially remove it either then or at some
+  later point.
+
+- **More discussion about SIGHASH_NOINPUT_UNSAFE:** Anthony Towns
+  started another [thread][noinput thread] about ensuring the proposed
+  noinput sighash mode is hard to misuse in ways that could result in
+  loss of funds.  Noinput can enable an alternative onchain enforcement
+  layer for LN that's more suitable for channels with multiple
+  participants, ultimately allowing a greater number of channels to be
+  opened in the same amount of block space.  Towns describes a
+  reasonably plausible worst-case scenario where failure to ensure
+  noinput safety puts adoption of other valuable protocol features at
+  risk.  To avoid that happening, he proposes refinements to earlier
+  ideas about output tagging (see [Newsletter #34][]) and a new
+  alternative that would require every transaction with a signature that
+  uses noinput to also contain a signature that doesn't use noinput.
+  This would prevent third-parties from executing replay attacks with
+  noinput transactions, but it would come with the downside of
+  moderately larger transaction sizes when noinput was being used.
+
```

Comment:
The participants in the script can share the private key for the extra signature, so they can create signatures on demand.  They'll need to store that private key (32 bytes) but won't need to store an extra signature per update, so the extra storage burden seems minimal to me.

The text says "moderately larger tranaction sizes when noinput was being used", so I'm not sure I understand your question there.  The taproot commitment will be the same size, but the revealed script will need to be larger.  The tree depth will probably be the same either way, so that doesn't change.  AJ covers all this in his email, so I think it's ok for us just to say "moderately larger transaction sizes" and let interested people follow the link.

Reply:
The extra signatures to generate and store off-chain for my eltoo variant is for extra privacy and to reduce on-chain costs; you could still just always do NOINPUT if you were happy for any unilateral channel close to give away that you were doing an eltoo channel. So in that sense I think harding's right.

Using a script is always larger than just spending with a key, and my preferred proposal requires two (non-aggregated) signatures and hence some key reveals as well, so definitely larger witness data -- that's why it's worth at least thinking about extra off-chain sigs.